tests/components/islamic_prayer_times/test_sensor.py

```diff
@@ -21,22 +22,45 @@ def set_utc(hass: HomeAssistant) -> None:
     hass.config.set_time_zone("UTC")
 
 
-async def test_islamic_prayer_times_sensors(hass: HomeAssistant) -> None:
+async def test_islamic_prayer_times_sensors(
+    hass: HomeAssistant, requests_mock: requests_mock.Mocker
+) -> None:
     """Test minimum Islamic prayer times configuration."""
     entry = MockConfigEntry(domain=DOMAIN, data={})
     entry.add_to_hass(hass)
 
-    with patch(
-        "prayer_times_calculator.PrayerTimesCalculator.fetch_prayer_times",
-        return_value=PRAYER_TIMES,
-    ), freeze_time(NOW):
+    prayer_times_json = load_fixture("prayer_times.json", "islamic_prayer_times")
+    prayer_times = json.loads(prayer_times_json)
+    new_prayer_times_json = load_fixture(
+        "new_prayer_times.json", "islamic_prayer_times"
+    )
+    new_prayer_times = json.loads(new_prayer_times_json)
+
+    requests_mock.register_uri(
+        "GET",
+        REQUEST_URL,
+        [
+            {"text": prayer_times_json},
+            {"text": new_prayer_times_json},
+        ],
+    )
```

Comment:
We normally recommend that you patch the library.
Why did you change this to patch `requests` instead?